sdk/core/azure-core/azure/core/rest/_aiohttp.py

```diff
@@ -23,14 +23,62 @@
 # IN THE SOFTWARE.
 #
 # --------------------------------------------------------------------------
-
+import collections.abc
 import asyncio
+from itertools import groupby
 from typing import AsyncIterator
 from multidict import CIMultiDict
 from . import HttpRequest, AsyncHttpResponse
 from ._helpers_py3 import iter_raw_helper, iter_bytes_helper
 from ..pipeline.transport._aiohttp import AioHttpStreamDownloadGenerator
 
+class _ItemsView(collections.abc.ItemsView):
+    def __init__(self, ref):
+        super().__init__(ref)
+        self._ref = ref
+
+    def __iter__(self):
+        for key, groups in groupby(self._ref.__iter__(), lambda x: x[0]):
+            yield tuple([key, ", ".join(group[1] for group in groups)])
+
+    def __contains__(self, item):
+        assert isinstance(item, (list, tuple))
+        assert len(item) == 2
+        for k, v in self.__iter__():
+            if item[0] == k and item[1] == v:
+                return True
+        return False
+
+    def __repr__(self):
+        return f"dict_items({list(self.__iter__())})"
+
+
+class _CIMultiDict(CIMultiDict):
+    """Dictionary with the support for duplicate case-insensitive keys."""
+
+    def __iter__(self):
+        return iter(self.keys())
```

Comment:
Not sure if I saw tests for this one? e.g. `assert list(headers) == [...]`

Reply:
good point, let me add that!